dynamiq/nodes/images/generation.py

```diff
@@ -0,0 +1,296 @@
+import base64
+import io
+from enum import Enum
+from typing import Any, Callable, ClassVar, Literal
+
+import httpx
+from pydantic import BaseModel, ConfigDict, Field, PrivateAttr
+
+from dynamiq.connections import AWS as AWSConnection
+from dynamiq.connections import AzureAI as AzureAIConnection
+from dynamiq.connections import Gemini as GeminiConnection
+from dynamiq.connections import OpenAI as OpenAIConnection
+from dynamiq.connections import VertexAI as VertexAIConnection
+from dynamiq.nodes import ErrorHandling
+from dynamiq.nodes.agents.exceptions import ToolExecutionException
+from dynamiq.nodes.node import ConnectionNode, NodeGroup, ensure_config
+from dynamiq.runnables import RunnableConfig
+from dynamiq.utils.logger import logger
+
+
+class ImageSize(str, Enum):
+    """Supported image sizes for generation."""
+
+    SIZE_256x256 = "256x256"
+    SIZE_512x512 = "512x512"
+    SIZE_1024x1024 = "1024x1024"
+    SIZE_1024x1792 = "1024x1792"
+    SIZE_1792x1024 = "1792x1024"
+
+
+class ImageResponseFormat(str, Enum):
+    """Response format for generated images."""
+
+    URL = "url"
+    B64_JSON = "b64_json"
+
+
+def create_image_file(image_bytes: bytes, index: int = 0, original_name: str | None = None) -> io.BytesIO:
+    """Create a properly configured BytesIO file from image bytes.
+
+    Args:
+        image_bytes: Raw image bytes.
+        index: Index for naming multiple images.
+        original_name: Optional original filename to base the new name on.
+                       If provided, will create names like "original_0.png", "original_1.png".
+                       If None, will use generic names like "image_0.png".
+
+    Returns:
+        BytesIO object with name and content_type attributes.
+    """
+    image_file = io.BytesIO(image_bytes)
+
+    extension_to_mime = {
```

Comment:
Please also move it up to file level constant, as we don't need to define this map on each func call